recipes/eigen/all/conanfile.py

```diff
@@ -30,7 +30,14 @@ def export_sources(self):
         export_conandata_patches(self)
 
     def configure(self):
-        self.license = "MPL-2.0" if self.options.MPL2_only else ("MPL-2.0", "LGPL-3.0-or-later")
+        if Version(self.version) < "5.0.0" and not self.options.MPL2_only:
+            self.license = ("MPL-2.0", "LGPL-3.0-or-later")
+            return
+
+        if Version(self.version) >= "5.0.0":
+            self.options.rm_safe("MPL2_only")
+        self.license = "MPL-2.0"
```

Comment:
I think this is more readable, wdyt?

```suggestion
        self.license = "MPL-2.0"  # MPL-2 only
        if Version(self.version) >= "5.0.0":
            del self.options.MPL2_only
        elif not self.options.MPL2_only:  # < 5.0.0
            self.license = ("MPL-2.0", "LGPL-3.0-or-later")
```